subsys/net/ip/net_context.c

```diff
@@ -1264,7 +1324,23 @@ int net_context_recv(struct net_context *context,
 	case IPPROTO_TCP:
 		ret = net_tcp_recv(context, cb, user_data);
 		break;
-
+#if defined(CONFIG_SOCKET_CAN)
+	case CAN_RAW:
+		if (context->conn_handler) {
+			return 0;
+		}
+		context->recv_cb = cb;
+		struct sockaddr local_addr = {
```

Comment:
you cannot declare a variable in the middle of a code block, always at the beginning of it. So have a new code block opened in your CAN_RAW.

Reply:
done